src/main/java/org/opensearch/jobscheduler/transport/ExtensionJobActionResponse.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.transport;
+
+import java.io.IOException;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.extensions.action.ExtensionActionResponse;
+
+/**
+ * Response from extension job action, converts response params to a byte array
+ *
+ */
+public class ExtensionJobActionResponse<T extends Writeable> extends ExtensionActionResponse {
+
+    /**
+     * Instantiates a new ExtensionJobActionResponse
+     *
+     * @param actionResponse the response object holding the action response parameters
+     * @throws IOException if serialization fails
+     */
+    public ExtensionJobActionResponse(T actionResponse) throws IOException {
+        super(convertResponseToBytes(actionResponse));
+    }
+
+    /**
+     * Takes in an object of type T that extends {@link Writeable} and converts the response to a byte array
+     *
+     * @param <T> a class that extends writeable
+     * @param actionResponse the action parameters to be serialized
+     * @throws IOException if serialization fails
+     * @return the byte array of the parameters
+     */
+    public static <T extends Writeable> byte[] convertResponseToBytes(T actionResponse) throws IOException {
```

Comment:
this duplicates the static method in `ExtensionJobActionRequest`.  Seems it could just be a static method on the `Writeable` interface itself if it's that useful.  Or at least at some common parent of `ExtensionActionRequest` and `ExtensionActionResponse`.

I'm still not convinced it has any benefit beyond the `Writeable` interface that lets it be deserialized somewhere. Where is this byte array handled?